File: de.unistuttgart.iste.sqa.mpw.hamstersimulator.java/core/src/main/java/de/unistuttgart/hamster/hamsterviewmodel/impl/GameViewPresenterImpl.java
package de.unistuttgart.hamster.hamsterviewmodel.impl;

import de.unistuttgart.hamster.hamsterviewmodel.GameViewModel;
import de.unistuttgart.hamster.hamsterviewmodel.GameViewPresenter;

import java.util.concurrent.Semaphore;

public class GameViewPresenterImpl extends GameViewPresenter {

	public GameViewPresenterImpl() {
		super(new Semaphore(1, true));
	}

	@Override
	public void playClicked() {
		throw new RuntimeException("not implemented");
	}

	@Override
	public void pauseClicked() {
		throw new RuntimeException("not implemented");
	}

	@Override
	public void undoClicked() {
		throw new RuntimeException("not implemented");
	}

	@Override
	public void redoClicked() {
		throw new RuntimeException("not implemented");
	}

	@Override
	public void textTyped(String text) {
		throw new RuntimeException("not implemented");
	}

	@Override
	public GameViewModel getViewModel() {
		throw new RuntimeException("not implemented");
	}

	@Override
	public void bind() {
		throw new RuntimeException("not implemented");
	}

}

File: de.unistuttgart.iste.sqa.mpw.karasimulator.cpp/core/src/impl/GameViewPresenterImpl.cpp
#include "GameViewPresenterImpl.h"

#include <stdexcept>

namespace viewmodel {

GameViewPresenterImpl::GameViewPresenterImpl(framework::Semaphore semaphore)
: (semaphore)
{
}

void GameViewPresenterImpl::playClicked() {
	throw std::runtime_error("not implemented");
}

void GameViewPresenterImpl::pauseClicked() {
	throw std::runtime_error("not implemented");
}

void GameViewPresenterImpl::undoClicked() {
	throw std::runtime_error("not implemented");
}

void GameViewPresenterImpl::redoClicked() {
	throw std::runtime_error("not implemented");
}

void GameViewPresenterImpl::textTyped(std::string text) {
	throw std::runtime_error("not implemented");
}

std::shared_ptr<viewmodel::GameViewModel> GameViewPresenterImpl::getViewModel() {
	throw std::runtime_error("not implemented");
}

void GameViewPresenterImpl::bind() {
	throw std::runtime_error("not implemented");
}

}

